src/components/Chart/AmpereChart.tsx

```diff
@@ -66,25 +117,41 @@ const AmpereChart = ({
     } = useSelector(chartState);
     const { samplingRunning } = useSelector(appState);
     const isRealTimePane = useSelector(isRealTimePaneSelector);
-    const sendTriggerLevel = level => dispatch(updateTriggerLevelAction(level));
-    const updateTriggerLevel = level =>
+    const sendTriggerLevel = (level: number) =>
+        dispatch(updateTriggerLevelAction(level));
+    const updateTriggerLevel = (level: number) =>
         dispatch(triggerLevelSetAction({ triggerLevel: level }));
 
+    const formatCurrent = (uA: number) =>
+        typeof uA === 'number'
+            ? unit(uA, 'uA')
+                  .format({ notation: 'auto', precision: 4 })
+                  .replace('u', '\u00B5')
+            : (undefined as never);
+
     const timestampToLabel = React.useCallback(
         (_usecs, index, array) => {
+            if (typeof _usecs !== 'number') {
+                return undefined as never;
+            }
             let usecs = _usecs;
-            if (triggerOrigin != null) {
-                usecs -= indexToTimestamp(triggerOrigin);
+            const timestampAtTriggerOrigin =
+                triggerOrigin == null ? null : indexToTimestamp(triggerOrigin);
+            if (timestampAtTriggerOrigin != null) {
```

Comment:
is `usecs` reassigned later? it can be declared as a const on this line instead, e.g.
`const usecs = _usecs - (timestampAtTriggerOrigin ?? 0);